tensorflow/python/ops/math_ops.py

```diff
@@ -966,6 +966,12 @@ def cast(x, dtype, name=None):
 
   Note casting nan and inf values to integral types has undefined behavior.
 
+  Note this operation can lead to a loss of precision when converting non-tensor
+  inputs such as native Python `float` and `complex` variables to `tf.float64`
+  or `tf.complex128` tensors, since the input is first converted to the
+  `float32` data type and then widened. For such cases, the proper solution is
```

Comment:
More than "For such cases", we recommend using `tf.convert_to_tensor` for _any_ non-tensor inputs instead of `tf.cast`.  That's the purpose of the function.